Product filters

mnath30 opened the pull request:
Added the functional filter and sorting section to the products page

Please review the following pages:

--> Helper folder:
      The helper folder contains all the helper functions which are used by the filter component to perform the sorting, filtering. The initialState file contains the state which is used during clearing all the filters. The compose function is used to compose all the passed functions to it from the context.

 1. compose.js
 2. filterbrands.js
 3. filterByPrice.js
 4. filterByRating.js
 5. initialState.js
 6. sorting.js

-- > reducer folder
      The reducer folder contains the function which is called by dispatch.
1. filterReducer.js

--> context folder
      The context folder contains the product context
1. ProductContext.jsx
 
--> Components folder
    The Filter component is used to call the dispatch on applying filters and sorting. The ProductNotFound component is used to show the error page when any product from that category is not available
1. Filter.jsx
2. ProductNotFound.jsx

 The filtered products are then passed and displayed in the Product.jsx which is present in the pages/Product/Product.jsx component.

mnath30 committed:
feat: Added the Filter and Sort functionality

Added the functional filter and sorting section to the products page

mnath30 committed:
refactor: Deleted the not required fetchData file

Deleted the fetchData file from the helper folder.

Manpreet-hub reviewed (approved):
Everything looks Great! Liked how you have structured the files.
Suggestion:  After applied the filters  show total number of products on the UI like: "Showing Products (5)".

sharathnair9999 commented on src/context/ProductContext.jsx:
```diff
@@ -1,26 +1,25 @@
-                setData(responseData.data.products)
-            }
-            catch{
-                console.log('There was some error in processing your request')
```
Instead of consoling the error you can give an error message or an alert

mnath30 replied:
Thankyou for your comment, I will remove the console log and replace it with an alert at the end

sharathnair9999 commented on src/reducer/filterReducer.jsx:
```diff
@@ -0,0 +1,39 @@
+
+        case "3_STARS_AND_ABOVE":
+            return {...state,rating:action.payload}
+
```
Instead of putting 5 cases for the same filter, you can do it within one case and compare according to the payload you give

mnath30 replied:
I have made the suggested changes.

sharathnair9999 reviewed (commented):
Suggested few changes

mnath30 committed:
feat: Added the functionality to display the product count

1. Added the functionality to display the product count displayed in the page
2. Refactored the product reducer and added a common type rating for the rating component

mnath30 committed:
refactor: Removed the commented lines from filterReducer

Removed the not required commented lines from filterReducer

mnath30 commented:
> Everything looks Great! Liked how you have structured the files. Suggestion: After applied the filters show total number of products on the UI like: "Showing Products (5)".

Thankyou for your comment, I have made the suggested change

sharathnair9999 reviewed (approved):
Now it looks clean. Good work
